### DISCUSSION TITLE
Select values from not existing

### DISCUSSION THREAD

**Discussion by efkeyti** [filamentphp/filament#3333] [Ideas]
Hi! (foreign alert)

When using select relationship, I want to add a new data if there is no data in the table. I know what I wrote was confusing. So let me show you with code :)

```php
Forms\Components\Select::make('feature_id')
  ->relationship('feature', 'feature', function ($query, $record) {
      $notIn = ProjectExtraFeature::query()->where('project_id', '??????')->pluck('feature_id');
      return $query->whereNotIn('id', $notIn);
  })
```

In this case I need the project_id. But I can't access this value. So how can I do this or does the filament have a feature that I don't know for existing relations?

**Comment by danharrin** [comment]
Like this? https://filamentphp.com/docs/2.x/forms/fields#creating-new-records

**Reply by efkeyti** [reply]
Each record only needs to be added once. I want the categories that have been added to the popup not to be listed.

In this case we have project and project features. I already added Synchronization. I don't want it listed again
![image](https://user-images.githubusercontent.com/110393099/182150403-a955d7de-3f12-4d57-bc7f-a620f366f035.png)

Example schema drawing to be more explanatory. If post have category, don't list again same category.
![image](https://user-images.githubusercontent.com/110393099/182150544-4087faae-022d-4376-8ef9-6d3f0c9885a6.png)

I'm still trying to improve my English. Sorry :)

**Reply by danharrin** [reply]
I think you're looking for `whereDoesntHave()`, check the Laravel docs

**Reply by efkeyti** [reply]
Ooh thank you! Problem solved.

**Reply by karintemi** [reply]
> Like this? https://filamentphp.com/docs/2.x/forms/fields#creating-new-records

@efkeyti Sorry to bother you, could you share me a snippet of how do you implement that method? I'm stuck in the same situation. 
Thanks in advance.

**Reply by karintemi** [reply]
> Ooh thank you! Problem solved.

Hello! I've got the same problem. Could you post de code to resolve this?
Thanks a lot
